### PR TITLE
added remove button in coop UI

### PR DESCRIPTION
**Pull request by changyc9928**

### CODE REVIEW

File: database/coop.py [hunk 1]
```diff
@@ -121,6 +121,27 @@ def book(self, tag, values):
...
+                self.talent.remove(val)
+            elif val in self.artifact:
+                self.artifact.remove(val)
+            elif val in self.trounce:
+                self.trounce.remove(val)
+            elif val in self.world_boss:
+                self.world_boss.remove(val)
+            else:
+                continue
+            ret += f"You're calcelling {val}\n"
```
Comment (BryanHo99) [inline database/coop.py]: Typo: cancelling

File: database/coop.py [hunk 2]
```diff
@@ -121,6 +121,27 @@ def book(self, tag, values):
...
+            elif val in self.talent:
+                self.talent.remove(val)
+            elif val in self.artifact:
+                self.artifact.remove(val)
+            elif val in self.trounce:
+                self.trounce.remove(val)
+            elif val in self.world_boss:
+                self.world_boss.remove(val)
+            else:
+                continue
```
Comment (BryanHo99) [inline database/coop.py]: Maybe instead of continue, you add onto the return string where val cannot be cancelled.

Comment (BryanHo99) [reply database/coop.py]: Also if there's a lot of if-else checking, I would personally try applying the open closed principle, but just for you to think about. :)

**Comment by BryanHo99** [comment]
Maybe if I feel like it, I can help out with some refactoring. Just let me know what needs improvement/assign me a ticket.

**Review by idea456 — APPROVED** [review]
LGTM

File: main.py [hunk 3]
```diff
@@ -366,19 +398,23 @@ async def wishHistory_test(ctx: commands.Context, uid: str):
...
     i = 0
     fig, axs = plt.subplots(2, 2)
     for banner, wish in wishes.items():
         msg = f"__{banner_name[banner]}__\n"
-        msg += "\n".join(list(map(lambda x: str(x), list(filter(lambda x: x.rarity == 4 or x.rarity == 5, wish)))))
-        msg += "\n\n" + f"You are currently {wish[-1].pity if len(wish) > 0 else 0} under pity."
+        msg += "\n".join(list(map(lambda x: str(x),
+                         list(filter(lambda x: x.rarity == 4 or x.rarity == 5, wish)))))
+        msg += "\n\n" + \
+            f"You are currently {wish[-1].pity if len(wish) > 0 else 0} under pity."
```
Comment (idea456) [inline main.py:410]: Maybe for readability we could do something like:
```
to_string = lambda x: str(x)
four_five_star_wishes = list(filter(lambda x: x.rarity == 4 or x.rarity == 5, wish)
msg += "\n".join(list(map(to_string, four_five_star_wishes))
```

File: database/coop.py [hunk 4]
```diff
@@ -121,6 +121,27 @@ def book(self, tag, values):
...
+                self.weapon.remove(val)
+            elif val in self.talent:
+                self.talent.remove(val)
+            elif val in self.artifact:
+                self.artifact.remove(val)
+            elif val in self.trounce:
+                self.trounce.remove(val)
+            elif val in self.world_boss:
+                self.world_boss.remove(val)
+            else:
```
Comment (idea456) [inline database/coop.py:139]: Maybe we can declare a collection or iterable class for all of these lists? The collection class could have methods to iterate and check for all of this stuff, and also add options something like the one you did in `dropdown_edit`

**Pull request merged**